app/uk/gov/hmrc/helptosavefrontend/controllers/BankAccountController.scala

```diff
@@ -68,17 +69,30 @@ class BankAccountController @Inject() (val helpToSaveService:     HelpToSaveServ
       withErrors ⇒
         Ok(views.html.register.bank_account_details(withErrors, selectEmailPage)),
       { bankDetails ⇒
-        checkIfAlreadyEnrolledAndDoneEligibilityChecks(htsContext.nino) {
-          session ⇒
-            sessionCacheConnector.put(session.copy(bankDetails = Some(bankDetails)))
-              .fold(
-                error ⇒ {
-                  logger.warn(s"Could not update session with bank details: $error")
-                  internalServerError()
-                },
-                _ ⇒ SeeOther(routes.RegisterController.checkDetails().url)
-              )
-        }
+        barsService.validate(bankDetails).map[Future[Result]] {
+          case Right(true) ⇒
+            checkIfAlreadyEnrolledAndDoneEligibilityChecks(htsContext.nino) {
+              session ⇒
+                val sortCodeWithHyphens = bankDetails.sortCode.grouped(2).mkString("-")
+                sessionCacheConnector.put(session.copy(bankDetails = Some(bankDetails.copy(sortCode = sortCodeWithHyphens))))
+                  .fold(
+                    error ⇒ {
+                      logger.warn(s"Could not update session with bank details: $error")
+                      internalServerError()
+                    },
+                    _ ⇒ SeeOther(routes.RegisterController.checkDetails().url)
+                  )
+            }
+
+          case Right(false) ⇒
+            val formWithErrors = BankDetails.giveBankDetailsForm().fill(bankDetails)
+              .withError("sortCodeBars", "check_your_sortcode_is_correct")
```

Comment:
it'd be more flexible for the future if the message key wasn't hard coded here. If we ever change the message keys then someone might forget to update it here.

Reply:
yeah, any better approach for this ?

Reply:
perhaps you may be able to add something to `FormOps` in `BankDetailsValidation` as that interface is being used already for the same purpose